### DISCUSSION TITLE
reality能跑到在vps上搭建的speedtest测出来的速度吗？

### DISCUSSION THREAD

**Discussion by lockqueen** [XTLS/Xray-core#2187] [Performance]
全局过xray环境下，speedtest.net测速上传下载1.5Mbps左右，
在vps上搭建的speedtest速度上下带宽50Mbps。
不确定这是否正常……

**Comment by lockqueen** [comment]
> 全局过xray环境下，speedtest.net测速上传下载1.5Mbps左右，
> 在vps上搭建的speedtest速度上下带宽50Mbps。
> 不确定这是否正常……

在vps上用speedtest-cli测速，上下带宽900Mbps……
猜测不到是哪里出了问题……

**Comment by chise0713** [comment]
你是使用的tcp、h2还是grpc，我也遇到过这个状况，我从h2或grpc换成tcp之后下载流量就正常了，tcp还可以加上xtls流控

**Reply by lockqueen** [reply]
用的是relity-xtls-vision-tcp，测速不太正常
passwall测试google连接延迟也是不太稳定
目前还不知道原因。。。变量太多不太能确定哪里有问题

**Reply by chise0713** [reply]
> 用的是relity-xtls-vision-tcp，测速不太正常 passwall测试google连接延迟也是不太稳定 目前还不知道原因。。。变量太多不太能确定哪里有问题

有套warp吗？可能是套的warp的问题？

**Reply by lockqueen** [reply]
没有吧……我是复制粘贴example，没做什么特别的设置。
刚刚又测速了一下，比上次多了50M了，speedtest.net下载有120M了
可能我没有完全掌握配置要领吧

**Reply by chise0713** [reply]
我的配置也是没办法跑到满速，可能确实是REALITY的问题吧？

**Reply by lockqueen** [reply]
可能是测速的时候，服务器同时发送接收，速度减半或者减更加多一些吧……

**Reply by chise0713** [reply]
之前mkcp+vmess我是可以跑满速的，应该和你说的这个没关系

**Reply by chise0713** [reply]
如果你所在的地区有sni白名单，你可以试试看自己签个证书用tls作为传输层安全

**Reply by lockqueen** [reply]
librespeed和xray都经过nginx sni分流后，速度一样了\(≧▽≦)/

**Reply by chise0713** [reply]
> librespeed和xray都经过nginx sni分流后，速度一样了(≧▽≦)/

是一直都可以满速吗？还是说时好时坏。我用自签证书也是不能满速，裸vmess+mkcp都可以跑满速，没懂

**Reply by lockqueen** [reply]
一直都是两者速度都一样了，speedtest.net比librespeed快点点

**Reply by chise0713** [reply]
这样啊，我也测试了下没带传输层安全的裸协议，也是跑不满。这应该不是xray的问题，而是我vps线路的问题，之前能跑满可能只是运气比较好。顺便问下你的vps是哪家买的（
